yt_dlp/extractor/rokfin.py

```diff
@@ -254,3 +422,91 @@ def _real_extract(self, url):
+            for search_result in metadata.get('results', []):
+                video_id_ind, video_type = self._TYPES.get(traverse_obj(search_result, ('content_type', 'raw')), (None, None))
+                video_id = traverse_obj(search_result, video_id_ind, expected_type=int_or_none)
+                if not video_id or not video_type:
+                    self.write_debug(msg=f'skipping {search_result}')
+                    continue
+                yield self.url_result(url=f'{self._BASE_URL}/{video_type}/{video_id}')
+        if not query:
+            return
+        query_data = {'query': query, 'page': {'size': 100}}
+        total_pages = None
+        for page_number in itertools.count(1):
+            query_data['page']['current'] = page_number
+            search_results = self._run_search_query(
+                data=query_data,
+                note='Downloading search results page %d%s' % (page_number, format_field(total_pages, template=' of ~%d') if total_pages and total_pages >= page_number else ''),
+                errnote='Unable to download search results page %d%s' % (page_number, format_field(total_pages, template=' of ~%d') if total_pages and total_pages >= page_number else ''))
+            total_pages = traverse_obj(search_results, ('meta', 'page', 'total_pages'), expected_type=int_or_none)
+            yield from get_video_data(search_results)
+            if not search_results.get('results'):
+                return
+
+    def _run_search_query(self, data, note, errnote):
+        data_bytes = json.dumps(data).encode('utf-8')
+        search_results = self._download_json(
+            self._search_engine_access_url, self._SEARCH_KEY, note=note, errnote=errnote, fatal=False,
+            encoding='utf-8', data=data_bytes, headers={'authorization': self._search_engine_access_key})
+        if search_results is not False:
+            return search_results
+        self.write_debug('updating access credentials')
+        self._search_engine_access_url = self._search_engine_access_key = None
+        self._downloader.cache.store(self._CACHE_SECTION_NAME, 'url', None)
+        self._downloader.cache.store(self._CACHE_SECTION_NAME, 'key', None)
+        self._search_engine_access_url, self._search_engine_access_key = self._get_search_engine_access_credentials()
+        return self._download_json(
+            self._search_engine_access_url, self._SEARCH_KEY, note=note, errnote=errnote, fatal=False,
+            encoding='utf-8', data=data_bytes, headers={'authorization': self._search_engine_access_key}) or {}
+
+    def _get_search_engine_access_credentials(self):
+        notfound_err_page = self._download_webpage('https://rokfin.com/discover', self._SEARCH_KEY, expected_status=404, fatal=False)
+        js_content = ''
+        search_engine_access_url = search_engine_access_key = None
+        for js_file_path in re.finditer(r'<script\s+[^>]*?src\s*=\s*"(?P<path>/static/js/[^">]*)"[^>]*>', notfound_err_page):
```

Comment:
```suggestion
        for js_file_path in re.finditer(r'<script[^>]+src\s*=\s*"(?P<path>/static/js/[^">]+)"', notfound_err_page):
```

Reply:
`<script[^>]+src\s*=` matches, e.g., `<scripttttttsrc=` which is not our intention. There should at least be one space following `<script`, right?

Reply:
You can use `\s` or `\b`:
```
r'<script\s[^>]*?\bsrc\s*=\s*"(?P<path>/static/js/[^">]+)"[^>]*>'
r'<script\b[^>]*?\bsrc\s*=\s*"(?P<path>/static/js/[^">]+)"[^>]*>'
```
No point using `\s+` as `[^>]` matches whitespace. Also, you presumably want the `path` group not to match just `/static/js/`, and to be the value of a `src` attribute rather than say `notasrc`; the above patterns match, eg, `... data-src="/static/js/abc.js" ...`: if you only want `... src=...` to match:
```
r'<script\b[^>]*?(?<!\w|-)src\s*=\s*"(?P<path>/static/js/[^">]+)"[^>]*>'
```
 Finally, you could also allow for single quotes:
```
r'''<script\b[^>]*?\bsrc\s*=\s*(?P<q>"|')(?P<path>/static/js/(?:(?!(?P=q))[^>])+)(?P=q)[^>]*>'''
```

Reply:
> <script[^>]+src\s*= matches, e.g., <scripttttttsrc= which is not our intention.

That is not valid html, but sure - you can keep the `\s`. It's more important to fix the `?` and `*`

Reply:
@pukkandan I don't see anything wrong with the question mark here. Should the corrected version be

`
'<script\s+[^>]*?\s+src\s*=\s*"(?P<path>/static/js/[^">]+)"\s+[^>]*>'
`

? This would match `'<script attr1 = "val1" attr2 src  = "/static/js/file.js" attr3="val3"  >'` but not `'<script attr1 = "val1" attr2src  = "/static/js/file.js" attr3="val3"  >'` and not `'<script attr1 = "val1" attr2 src  = "/static/js/" attr3="val3"  >'`.

If you want, I can remove the trailing `'\s+[^>]*>'` part which I only added for robustness sake.

Reply:
>  I don't see anything wrong with the question mark here.

And "nothing wrong" is good enough reason to add useless operators?

> '<script\s+[^>]*?\s+src\s*=\s*"(?P<path>/static/js/[^">]+)"\s+[^>]*>'

https://regex101.com/r/s8Vekc/1

```suggestion
        for js_file_path in re.finditer(r'<script\b[^>]*\ssrc\s*=\s*"(?P<path>/static/js/[^">]+)"', notfound_err_page):
```